include/fs/littlefs.h

```diff
@@ -17,6 +17,26 @@
 extern "C" {
 #endif
 
+typedef int (*lfs_device_open)(void **context, uintptr_t area_id);
```

Comment:
Shouldn't the `context` contain area_id and flash_area, `lfs_device_open()` should be general for all devices.

Reply:
The context is getting set by lfs_device_open, based upon the area id. In the flash case this is getting set to an instance of struct flash_area. The context later on is handed over to each of these device functions, as it basically contains the necessary information for the other flash operations. As it is a general void* it can be used differently, depending on the underlying device type and methods. I basically stole this idea from the previous implementation and generalized it a bit.
Hence, lfs_device_open cannot be general for all devices, but instead must be specific for the device type.